mobile-app/app/contexts/CustomServiceProvider.tsx

```diff
@@ -150,26 +152,28 @@ export function CustomServiceProvider(
   });
 
   useEffect(() => {
+    const url = isEvmFeatureEnabled ? evmUrl : defaultEvmUrl;
     setCurrentUrl((prevState) => ({
       ...prevState,
-      [CustomServiceProviderType.EVM]: evmUrl,
+      [CustomServiceProviderType.EVM]: url,
     }));
-  }, [evmUrl]);
+  }, [evmUrl, isEvmFeatureEnabled]);
 
   useEffect(() => {
+    const url = isEvmFeatureEnabled ? ethRpcUrl : defaultEthRpcUrl;
     setCurrentUrl((prevState) => ({
       ...prevState,
-      [CustomServiceProviderType.ETHRPC]: ethRpcUrl,
+      [CustomServiceProviderType.ETHRPC]: url,
     }));
-  }, [ethRpcUrl]);
+  }, [ethRpcUrl, isEvmFeatureEnabled]);
 
   const isCustomEvmUrl = useMemo(
-    () => currentUrl.EVM !== defaultEvmUrl,
-    [currentUrl.EVM, defaultEvmUrl],
+    () => isEvmFeatureEnabled && currentUrl.EVM !== defaultEvmUrl,
```

Comment:
can we skip this as above if `isEvmFeatureEnabled` is false then we are setting `evmUrl` to `defaultEvmUrl` only?

Reply:
Nice catch! Yep, tried without this extra condition and still works fine. Updated 👍